src/Microsoft.AspNetCore.Mvc.Razor.Precompilation/Program.cs

```diff
@@ -0,0 +1,69 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Linq;
+using System.Reflection;
+using Microsoft.AspNetCore.Mvc.Razor.Precompilation.Internal;
+
+namespace Microsoft.AspNetCore.Mvc.Razor.Precompilation
+{
+    public class Program
+    {
+        private readonly static Type ProgramType = typeof(Program);
+
+        public static int Main(string[] args)
+        {
+#if DEBUG
+            DebugHelper.HandleDebugSwitch(ref args);
+#endif
+
+            var app = new PrecompilationApplication(ProgramType);
+
+            EnsureValidDispatchRecipient(ref args);
+
+            PrecompileCommandBase.Register<PrecompileRunCommand>(app);
+
+            return app.Execute(args);
+        }
+
+        private static void EnsureValidDispatchRecipient(ref string[] args)
```

Comment:
I only did this in RazorTooling to limit the number of dependencies the inside man had. If you end up having to have cli.utils in your package graph for the inside man after the suggestions I've made above then you can use the DotnetToolDispatcher.EnsureValidDispatchRecipient